appeals/web/src/server/appeals/appeal-details/representations/common/components/reject-reasons.js

```diff
@@ -1,18 +1,68 @@
+import { ensureArray } from '#lib/array-utilities.js';
 import { buildHtmUnorderedList } from '#lib/nunjucks-template-builders/tag-builders.js';
 
 /**
- * @param {string[]} selectedReasons
- * @param {import('@pins/appeals.api').Appeals.RepresentationRejectionReason[]} rejectionReasons
- * @returns {string}
+ * @typedef {import('@pins/appeals.api').Appeals.ReasonOption} ReasonOption
+ */
+
+/**
+@param {Object} rejectComment
+@param {string|string[]} reasons
+@param {ReasonOption[]} reasonOptions
+ * @returns {string[]}
  * */
-export const rejectionReasonHtml = (selectedReasons, rejectionReasons) => {
-	const reasonNames = selectedReasons.map(
-		(reasonId) => rejectionReasons.find((r) => r.id === parseInt(reasonId))?.name ?? ''
-	);
+export const prepareRejectionReasons = (rejectComment, reasons, reasonOptions) => {
+	const reasonsText = reasonOptions
+		.filter((reasonOption) => reasonOption.hasText)
+		.reduce((acc, { id }) => {
```

Comment:
may as well just move the filter condition into the reduce as a guard clause and save a loop